packages/grafana-data/src/types/explore.ts

```diff
@@ -10,11 +11,20 @@ export interface ExploreUrlState<T extends DataQuery = AnyQuery> {
   range: RawTimeRange;
   originPanelId?: number;
   context?: string;
+  panelsState?: ExplorePanelsState;
+}
+
+export interface ExplorePanelsState extends Partial<Record<PreferredVisualisationType, {}>> {
```

Comment:
If you want to add support for other data visualizations to serialize state into the url you do it here. For the "graph type example", you could do...

```
export interface ExplorePanelsState extends Partial<Record<PreferredVisualisationType, {}>> {
  trace?: ExploreTracePanelState;
  graph?: ExploreGraphPanelState;
}

export interface ExploreGraphPanelState {
  type: string;
}
```

The state will now be available in redux like so:
`useSelector((state: StoreState) => state.explore[props.exploreId]?.panelsState.graph);`

And can be changed like so:
```
  const dispatch = useDispatch();
  dispatch(changePanelState(props.exploreId, 'graph', { type: 'asdf' }));
```